packages/@aws-cdk/aws-dynamodb-global/lib/aws-dynamodb-global.ts

```diff
@@ -0,0 +1,71 @@
+import dynamodb = require("@aws-cdk/aws-dynamodb");
+import cdk = require("@aws-cdk/cdk");
+import { LambdaGlobalDynamoDBMaker } from "./lambda-global-dynamodb";
+import { MultiDynamoDBStack } from "./multi-dynamodb-stack";
+
+export interface GlobalDynamoDBProps extends dynamodb.TableOptions {
+    /**
+     * Enforces a particular table name
+     * to share across Global DynamoDB tables
+     */
+    tableName: string;
+}
+
+/**
+ * Properties for the mutliple DynamoDB tables to mash together into a
+ * global table
+ */
+export interface GlobalTableProps extends cdk.StackProps {
+    /**
+     * Properties for DynamoDB Tables
+     * All the properties across tables/regions must be exactly the same
+     * to make the tables mesh together as a global table
+     */
+    dynamoProps: GlobalDynamoDBProps;
+
+    /**
+     * Array of environments to create DynamoDB tables in
+     * Accounts should be omitted, or at least all identical
+     */
+    regions: string[];
+}
+
+export class GlobalTable extends cdk.Construct {
+
+    /**
+     * Creates dynamoDB tables across regions that will be able to be globbed together into a global table
+     */
+    public tables: MultiDynamoDBStack[];
```

Comment:
I think the API should only expose the `Table` objects and not the stack that wraps them. Furthermore, I don't even thing you need this `MultiDynamoDBStack` class (and of course you don't need it in the public API). You can just create those stacks in-place:

```ts
for (const reg of props.regions) {
  const regionalStack = new cdk.Stack(this, '...', { env: { region: reg } });
  const regionalTable = new dynamodb.Table(regionalStack, 'Table', props);
  this.tables.push(regionalTable);
}
```

This is how you should define the property:

```ts
public readonly tables = new Array<dynamodb.Table>();
```

P.S. this is not ideal because the array is mutable. The "right" thing to do would be to return a clone:

```ts
private readonly _regionalTables = new Array<dynamodb.Table>();

public get regionalTables() {
  return this._regionalTables.map(x => x);
}
```

Reply:
Chiming in—if mutability is a concern, would it make sense to use a library like https://github.com/immutable-js/immutable-js, or would that be too big of a dependency to expose as a public API in the CDK?

Reply:
We prefer to avoid taking such dependencies (and frankly JS/TS are sufficient). Soon Typescript will have `x: readonly string[]` and the world will be whole.